src/python/pants/backend/python/providers/python_build_standalone/rules.py

```diff
@@ -0,0 +1,244 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+from __future__ import annotations
+
+import collections.abc
+import functools
+import json
+import textwrap
+from typing import Iterable, TypedDict, cast
+
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.util_rules.interpreter_constraints import InterpreterConstraints
+from pants.backend.python.util_rules.pex import PythonProvider
+from pants.backend.python.util_rules.pex import rules as pex_rules
+from pants.backend.python.util_rules.pex_environment import PythonExecutable
+from pants.core.util_rules.external_tool import (
+    DownloadedExternalTool,
+    ExternalToolError,
+    ExternalToolRequest,
+)
+from pants.core.util_rules.external_tool import rules as external_tools_rules
+from pants.core.util_rules.system_binaries import CpBinary
+from pants.engine.fs import DownloadFile
+from pants.engine.internals.native_engine import FileDigest
+from pants.engine.internals.selectors import Get
+from pants.engine.platform import Platform
+from pants.engine.process import Process, ProcessCacheScope, ProcessResult
+from pants.engine.rules import collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.option.global_options import NamedCachesDirOption
+from pants.option.option_types import StrListOption
+from pants.option.subsystem import Subsystem
+from pants.util.docutil import bin_name
+from pants.util.frozendict import FrozenDict
+from pants.util.logging import LogLevel
+from pants.util.resources import read_sibling_resource
+from pants.util.strutil import softwrap
+
+PBS_SANDBOX_NAME = ".python_build_standalone"
+PBS_NAMED_CACHE_NAME = "python_build_standalone"
+PBS_APPEND_ONLY_CACHES = FrozenDict({PBS_NAMED_CACHE_NAME: PBS_SANDBOX_NAME})
+
+
+class PBSPythonInfo(TypedDict):
+    url: str
+    sha256: str
+    size: int
+
+
+@functools.cache
+def load_pbs_pythons() -> dict[str, dict[str, PBSPythonInfo]]:
+    return cast(
+        "dict[str, dict[str, PBSPythonInfo]]",
+        json.loads(read_sibling_resource(__name__, "versions_info.json"))["pythons"],
+    )
+
+
+class PBSPythonProviderSubsystem(Subsystem):
+    options_scope = "pbs-python-provider"
```

Comment:
What color should this shed be?  `pbs-python-provider` or `python-build-standalone-python-provider`?

Reply:
PBS doesn't seem to refer to itself as PBS ever, e.g. in the repo or docs:

- https://github.com/search?q=repo%3Aindygreg%2Fpython-build-standalone%20PBS&type=code
- https://gregoryszorc.com/docs/python-build-standalone/main/search.html?q=PBS

So, I'd be inclined to the second: laboriously spelling it out to match the tool rather than impose pants-isms. (Fine to use the abbreviation in the doc string where we can explain it, though.)